packages/ui-scripts/lib/build/buildThemes/setupThemes.js

```diff
@@ -109,19 +110,44 @@ const setupThemes = async (targetPath, input) => {
           `
     await createFile(`${themePath}/semantics.ts`, semanticsFileContent)
 
+    const componentAndSubcomponentNames = []
     //components
     for (const componentpath of themeData[theme].components) {
       const rawComponentName =
         componentpath.split('/')[componentpath.split('/').length - 1]
-      const componentName =
+      const mainComponentName =
         rawComponentName[0].toLowerCase() + rawComponentName.slice(1)
 
-      const component = generateComponent(input[componentpath][componentName])
-      const componentTypes = generateComponentType(
-        input[componentpath][componentName]
+      //figma needs prefixed subcomponents, e.g.: for modal => modalHeader, modalBody, etc... and we need only the name of the subcomponent, so we access the data with the subcomponent object, containing the prefixed version and making a non-prefixed list of the same components to produce files, variable, theme names
+      const subcomponents = Object.keys(input[componentpath])
+      const subcomponentNames = subcomponents.reduce(
+        (acc, subcomponent) => [
+          ...acc,
+          ...(subcomponent.startsWith(acc[0]) &&
+          subcomponent.length > acc[0].length
+            ? [unCapitalize(subcomponent.slice(acc[0].length))]
+            : [])
+        ],
+        [subcomponents[0]]
       )
 
-      const componentFileContent = `
+      subcomponents.forEach(async (subcomponent, subcomponentIndex) => {
```

Comment:
You should use here a `for...of`, so the return value of the promises is not ignored